x-pack/filebeat/input/awss3/input.go

```diff
@@ -199,15 +199,25 @@ func (in *s3Input) createS3Lister(ctx v2.Context, cancelCtx context.Context, cli
 		s3ServiceName = "s3-fips"
 	}
 
-	s3Client := s3.New(awscommon.EnrichAWSConfigWithEndpoint(in.config.AWSConfig.Endpoint, s3ServiceName, in.awsConfig.Region, in.awsConfig))
-	regionName, err := getRegionForBucketARN(cancelCtx, s3Client, in.config.BucketARN)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get AWS region for bucket_arn: %w", err)
+	var s3Client *s3.Client
+	endpoint, _ := url.Parse(in.config.AWSConfig.Endpoint)
```

Comment:
I would move this logic to `awscommon.EnrichAWSConfigWithEndpoint` returning directly either the composed endpoint or the raw one if it's a full URI